7 ftp using the universal mounting system (#31)
opulo-inc/docs: content/en/openpnp/calibration/homing-fiducial/index.md

  Mounting datum board and testing the homing fiducial
---

> Now that we've done the fisheye calibration, we can mount the datum board to the staging plate. The datum board is a reference point for everything else you'll do with the machine. The center dot of the datum board will be the most important calibration point for the machine. It'll be used to fine-tune the LumenPnP's XY position after homing to account for any errors in the limit switches.

Reviewer: ```suggestion
Now that fisheye calibration is complete, you can mount the datum board to the staging plate. The datum board is a reference point for everything else you'll do with the machine. The center dot of the datum board will be the most important calibration point for the machine. It'll be used to fine-tune the LumenPnP's XY position after homing to account for any errors in the limit switches.
```

1. Use four M3x16mm screws and four M3 nuts to secure the datum board and datum board mount to the staging plate on the rear of the bottom camera, through holes: B18, A19, A21, B22. The the fisheye calibration pattern should be facing down, and the gold grid of lines and fiducial in the center of the Opulo logo facing upwards. Tighten this down securely.
   {{< container-image path="images/Screen Shot 2022-01-11 at 12.51.47 PM.png" alt="the mounting location for the datum board and holder" >}}

2. In the "Machine Controls" pane in the lower left, connect to the LumenPnP if you haven't already by pressing the power button.

> 3. Run a rough home routine by pressing the Home button (shaped like a house). The machine will move eto the X and Y axis limit switches. The default machine configuration files are set so that OpenPnP will *not* try to use the homing fiducial yet. Note that the home icon will turn yellow, as the LumenPnP has homed to it's end stops, but hasn't completed its full homing routine.

Reviewer: lil typo

```suggestion
3. Run a rough home routine by pressing the Home button (shaped like a house). The machine will move to the X and Y axis limit switches. The default machine configuration files are set so that OpenPnP will *not* try to use the homing fiducial yet. Note that the home icon will turn yellow, as the LumenPnP has homed to it's end stops, but hasn't completed its full homing routine.
```

{{< container-image path="images/Screen Shot 2022-05-18 at 2.27.08 PM.png" alt="" >}}

> 12. Lastly, we can confirm that our edits result in accurate fiducial detection. Once again click on the row with "DrawCircles" in the Stage column. The image will show you every circle detected from the pipeline. Confirm there is only one, and it is perfectly hugging the fiducial as shown below.

Reviewer: ```suggestion
12. Lastly, you can confirm that the changes result in accurate fiducial detection. Once again click on the row with "DrawCircles" in the Stage column. The image will show you every circle detected from the pipeline. Confirm there is only one, and it is perfectly hugging the fiducial as shown below.
```

{{< container-image path="images/Screen Shot 2022-05-18 at 2.41.29 PM.png" alt="" >}}

13. Exit out of the pipeline editing UI, save your changes, and try to home again. With your vision pipeline tuned, the machine should find the homing fiducial on your machine and center the crosshairs of the top camera on it.

> 4. It is likely that the machine will move to the wrong place and not be able to identify the homing fiducial. If this happens, OpenPnP will give an error popup that says `FIDUCIAL-HOME no matches found`. This means we'll need to fine-tune the location that OpenPnP moves the LumenPnP to in order to find the homing fiducial. (this is normal). You'll also note that the home icon will turn yellow, as the LumenPnP has homed to it's end stops, but hasn't completed its full homing routine.

Reviewer: ```suggestion
4. It is likely that the machine will move to the wrong place and not be able to identify the homing fiducial. If this happens, OpenPnP will give an error popup that says `FIDUCIAL-HOME no matches found`. This is normal, as it means you'll need to fine-tune the location that OpenPnP moves the LumenPnP to in order to find the homing fiducial. You'll also note that the home icon will turn yellow, as the LumenPnP has homed to the end stops, but hasn't completed the full homing routine.
```
